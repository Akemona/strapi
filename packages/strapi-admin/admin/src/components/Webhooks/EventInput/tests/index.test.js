import React from 'react';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import translationMessages from '../../../../translations/en.json';
import EventInput from '../index';
import EventRow from '../EventRow';

const intlProvider = new IntlProvider({ locale: 'en', message: translationMessages }, {});
const { intl } = intlProvider.state;

function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

const shallowWithIntl = node => {
  return shallow(nodeWithIntlProp(node), { context: { intl } });
};

describe('<EventInput />', () => {
  const props = {
    name: 'events',
    value: ['media.create', 'media.delete'],
    onChange: jest.fn(),
    shouldShowDPEvents: false,
  };

  it('should match the snapshot', () => {
    const tree = renderer.create(
      <IntlProvider locale="en" messages={translationMessages} textComponent="span">
        <EventInput {...props} />
      </IntlProvider>
    );

    expect(tree).toMatchSnapshot();
  });

  describe('OnChange prop called with right params', () => {
    it('should add an event on handleChange if value is true', () => {
      const event = {
        target: {
          name: 'entry.update',
          value: true,
        },
      };

      const formattedEvent = {
        target: {
          name: 'events',
          value: ['media.create', 'media.delete', 'entry.update'],
        },
      };

      let wrapper = shallowWithIntl(<EventInput {...props} />);
      wrapper.dive();
      let child = wrapper.find(EventRow);

      child
        .at(0)
        .props()
        .handleChange(event);

      expect(props.onChange).toHaveBeenCalledWith(formattedEvent);
    });

    it('should delete an event on handleChange if value is false', () => {
      const event = {
        target: {
          name: 'media.create',
          value: false,
        },
      };

      const formattedEvent = {
        target: {
          name: 'events',
          value: ['media.delete'],
        },
      };

      let wrapper = shallowWithIntl(<EventInput {...props} />);
      wrapper.dive();
      let child = wrapper.find(EventRow);

      child
        .at(0)
        .props()
        .handleChange(event);

      expect(props.onChange).toHaveBeenCalledWith(formattedEvent);
    });
  });

  it('should add all events from the same category on handleChangeAll if value is true', () => {
    const event = {
      target: {
        name: 'entry',
        value: true,
      },
    };

    const formattedEvent = {
      target: {
        name: 'events',
        value: ['media.create', 'media.delete', 'entry.create', 'entry.update', 'entry.delete'],
      },
    };

    let wrapper = shallowWithIntl(<EventInput {...props} />);
    wrapper.dive();
    let child = wrapper.find(EventRow);

    child
      .at(0)
      .props()
      .handleChangeAll(event);

    expect(props.onChange).toHaveBeenCalledWith(formattedEvent);
  });

  it('should delete all events from the same category on handleChangeAll if value is false', () => {
    const event = {
      target: {
        name: 'media',
        value: false,
      },
    };

    const formattedEvent = {
      target: {
        name: 'events',
        value: [],
      },
    };

    let wrapper = shallowWithIntl(<EventInput {...props} />);
    wrapper.dive();
    let child = wrapper.find(EventRow);

    child
      .at(0)
      .props()
      .handleChangeAll(event);

    expect(props.onChange).toHaveBeenCalledWith(formattedEvent);
  });
});
