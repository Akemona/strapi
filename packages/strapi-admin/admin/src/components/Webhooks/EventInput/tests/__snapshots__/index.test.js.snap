// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventInput /> should match the snapshot 1`] = `
.c2 {
  cursor: pointer;
  margin: 0;
  position: relative;
  width: 14px;
}

.c2:focus,
.c2:active {
  outline: 0;
}

.c2:before {
  content: '';
  position: absolute;
  left: 0;
  top: 50%;
  margin-top: calc(-14px / 2);
  width: 14px;
  height: 14px;
  border: 1px solid rgba(16,22,34,0.15);
  background-color: #fdfdfd;
  border-radius: 3px;
  box-sizing: border-box;
}

.c2:after {
  display: none;
  content: '\\f00c';
  font-family: 'FontAwesome';
  position: absolute;
  left: 0;
  top: 55%;
  margin-top: calc(-14px / 2);
  width: 14px;
  height: 14px;
  text-align: center;
  font-size: 9px;
  color: #1C5DE7;
  line-height: 14px;
}

.c2:checked:after {
  display: block;
}

.c2:disabled {
  background-color: #FAFAFB;
  cursor: initial;
}

.c2:disabled:after {
  color: #9ea7b8;
}

.c2 + label {
  display: inline-block;
  font-weight: 500;
  font-size: 1.3rem;
}

.c4 {
  cursor: pointer;
  margin: 0;
  position: relative;
  width: 14px;
}

.c4:focus,
.c4:active {
  outline: 0;
}

.c4:before {
  content: '';
  position: absolute;
  left: 0;
  top: 50%;
  margin-top: calc(-14px / 2);
  width: 14px;
  height: 14px;
  border: 1px solid rgba(16,22,34,0.15);
  background-color: #fdfdfd;
  border-radius: 3px;
  box-sizing: border-box;
}

.c4:after {
  display: none;
  content: '\\f00c';
  font-family: 'FontAwesome';
  position: absolute;
  left: 0;
  top: 55%;
  margin-top: calc(-14px / 2);
  width: 14px;
  height: 14px;
  text-align: center;
  font-size: 9px;
  color: #1C5DE7;
  line-height: 14px;
}

.c4:checked:after {
  display: block;
}

.c4:after {
  content: '\\f068';
  display: block;
  top: 50%;
  font-size: 10px;
}

.c4:disabled {
  background-color: #FAFAFB;
  cursor: initial;
}

.c4:disabled:after {
  color: #9ea7b8;
}

.c4 + label {
  display: inline-block;
  font-weight: 500;
  font-size: 1.3rem;
}

.c1 input,
.c1 label {
  display: inline-block;
  vertical-align: top;
}

.c1 input {
  margin-top: 4px;
  width: 14px;
  height: 12px;
}

.c1 input + label {
  display: inline-block;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  max-width: calc(100% - 14px);
  margin-left: 10px;
  margin-bottom: 0;
  line-height: 18px;
  cursor: pointer;
}

.c3 {
  width: 100%;
  font-weight: 500;
  font-size: 1.3rem;
  color: #333740;
}

.c0 {
  padding-top: 3px;
  padding-bottom: 8px;
}

.c0 table {
  width: 100%;
  border-radius: 2px;
  overflow: hidden;
}

.c0 tr:before {
  content: '-';
  position: absolute;
  left: 20px;
  display: inline-block;
  width: calc(100% - 40px);
  height: 1px;
  margin-top: -1px;
  line-height: 1.1em;
  color: transparent;
  background-color: #f6f6f6;
}

.c0 tr:first-of-type:before {
  left: 0;
  height: 2px;
  width: 100%;
  background-color: #fafafb;
  z-index: 1;
}

.c0 thead {
  background-color: #fafafb;
}

.c0 thead td {
  height: 41px;
  text-align: center;
  font-size: 13px;
  font-weight: 500;
}

.c0 tbody {
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  box-shadow: inset 0px 0px 0px 1px #f6f6f6;
}

.c0 tbody td {
  height: 54px;
  padding-top: 3px;
}

.c0 tbody td input[type='checkbox']:disabled:before {
  background-color: #e8e5e5;
}

.c0 tbody td:first-of-type {
  width: 200px;
  padding-left: 30px;
  text-transform: capitalize;
}

.c0 tbody td:first-of-type label {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: 500;
  cursor: pointer;
}

.c0 tbody td:not(:first-of-type) > div {
  display: block;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin: 0 auto;
}

<div
  className="c0"
>
  <table>
    <thead>
      <tr>
        <td />
        <td>
          <span>
            Create
          </span>
        </td>
        <td>
          <span>
            Update
          </span>
        </td>
        <td>
          <span>
            Delete
          </span>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={false}
              className="c2"
              disabled={false}
              id="entry"
              name="entry"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="entry"
            >
              entry
            </label>
          </div>
        </td>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={false}
              className="c2"
              disabled={false}
              id="entry.create"
              name="entry.create"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="entry.create"
            />
          </div>
        </td>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={false}
              className="c2"
              disabled={false}
              id="entry.update"
              name="entry.update"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="entry.update"
            />
          </div>
        </td>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={false}
              className="c2"
              disabled={false}
              id="entry.delete"
              name="entry.delete"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="entry.delete"
            />
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={false}
              className="c4"
              disabled={false}
              id="media"
              name="media"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="media"
            >
              media
            </label>
          </div>
        </td>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={true}
              className="c2"
              disabled={false}
              id="media.create"
              name="media.create"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="media.create"
            />
          </div>
        </td>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={false}
              className="c2"
              disabled={false}
              id="media.update"
              name="media.update"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="media.update"
            />
          </div>
        </td>
        <td>
          <div
            className="c1"
            disabled={false}
          >
            <input
              autoComplete="off"
              autoFocus={false}
              checked={true}
              className="c2"
              disabled={false}
              id="media.delete"
              name="media.delete"
              onChange={[Function]}
              placeholder={null}
              tabIndex="0"
              type="checkbox"
            />
            <label
              className="c3"
              htmlFor="media.delete"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
